infra/build/functions/test_data/expected_centipede_build_steps.json

```diff
@@ -0,0 +1,308 @@
+      "."
+    ],
+    "dir": "oss-fuzz/projects/test-project"
+  },
+  {
+    "name": "gcr.io/oss-fuzz/test-project",
+    "args": [
+      "bash",
+      "-c",
+      "srcmap > /workspace/srcmap.json && cat /workspace/srcmap.json"
+    ],
+    "env": [
+      "OSSFUZZ_REVISION=$REVISION_ID",
+      "FUZZING_LANGUAGE=c++"
+    ],
+    "id": "srcmap"
+  },
+  {
+    "name": "gcr.io/cloud-builders/docker",
+    "env": [
+      "ARCHITECTURE=x86_64",
+      "FUZZING_ENGINE=centipede",
+      "FUZZING_LANGUAGE=c++",
+      "HOME=/root",
+      "OUT=/workspace/out/centipede-address-x86_64",
+      "SANITIZER=address"
+    ],
+    "args": [
+      "run",
+      "--platform",
+      "linux/amd64",
+      "-v",
+      "/workspace:/workspace",
+      "-e",
+      "ARCHITECTURE=x86_64",
+      "-e",
+      "FUZZING_ENGINE=centipede",
+      "-e",
+      "FUZZING_LANGUAGE=c++",
+      "-e",
+      "HOME=/root",
+      "-e",
+      "OUT=/workspace/out/centipede-address-x86_64",
+      "-e",
+      "SANITIZER=address",
+      "-t",
+      "gcr.io/oss-fuzz/test-project",
+      "bash",
+      "-c",
+      "rm -r /out && cd /src && cd /src && mkdir -p /workspace/out/centipede-address-x86_64 && compile || (echo \"********************************************************************************\nFailed to build.\nTo reproduce, run:\npython infra/helper.py build_image test-project\npython infra/helper.py build_fuzzers --sanitizer address --engine centipede --architecture x86_64 test-project\n********************************************************************************\" && false)"
+    ],
+    "id": "compile-centipede-address-x86_64"
+  },
+  {
+    "name": "gcr.io/cloud-builders/docker",
+    "env": [
+      "ARCHITECTURE=x86_64",
+      "FUZZING_ENGINE=centipede",
+      "FUZZING_LANGUAGE=c++",
+      "HOME=/root",
+      "OUT=/workspace/out/centipede-address-x86_64",
+      "SANITIZER=address"
+    ],
+    "args": [
+      "run",
+      "--platform",
+      "linux/amd64",
+      "-v",
+      "/workspace:/workspace",
+      "-e",
+      "ARCHITECTURE=x86_64",
+      "-e",
+      "FUZZING_ENGINE=centipede",
+      "-e",
+      "FUZZING_LANGUAGE=c++",
+      "-e",
+      "HOME=/root",
+      "-e",
+      "OUT=/workspace/out/centipede-address-x86_64",
+      "-e",
+      "SANITIZER=address",
+      "-t",
+      "gcr.io/oss-fuzz-base/base-runner",
+      "bash",
+      "-c",
+      "test_all.py || (echo \"********************************************************************************\nBuild checks failed.\nTo reproduce, run:\npython infra/helper.py build_image test-project\npython infra/helper.py build_fuzzers --sanitizer address --engine centipede --architecture x86_64 test-project\npython infra/helper.py check_build --sanitizer address --engine centipede --architecture x86_64 test-project\n********************************************************************************\" && false)"
+    ],
+    "id": "build-check-centipede-address-x86_64"
```

Comment:
Will bad-build-check still work as is for this? Since we are building sanitizer vs non-sanitizer independently, this will fail right? 

For now, we still want to keep the sanitizer instrumentation checks, just not run the fuzzers.  Longer term, we need to fix bad-build-check such that it properly runs with sanitizers with this model.

Reply:
Has it been discussed what the model for centipede builds should be?
I think the easiest model to fit into the existing ones is to rebuild the project twice (once with ASAN, once with nothing) when doing an ASAN build. This would break on projects without build.sh but I think it fits most neatly into CF's concept of builds.

Reply:
> Has it been discussed what the model for centipede builds should be?

My understanding is that we have two separate builds, one for sanitized and the other for unsanitized.
I am not familiar with `OSS-Fuzz` enough to tell which one is better, but I suppose when `Centipede` supports multiple sanitizers in the future, we don't need it to build an unsanitized binary for each sanitized one?

Reply:
Yes, the plan here is to have separate builds for unsanitized + sanitized binaries.